types/user/usergroups.d.ts

```diff
@@ -1,40 +1,62 @@
+/// <reference path="../shared.d.ts" />
+
 declare module 'libsession_util_nodejs' {
   /**
    *
    * UserGroups wrapper logic
    *
    */
 
-  export type UserGroupsType = 'Community' | 'LegacyGroup';
+  export type UserGroupsType = 'Community' | 'LegacyGroup' | 'Group';
 
   export type CommunityDetails = {
     fullUrlWithPubkey: string;
     baseUrl: string;
     roomCasePreserved: string;
   };
 
-  export type CommunityInfo = CommunityDetails & {
-    pubkeyHex: string;
-    priority: number; // -1 means hidden, 0 means normal, > 0 means pinned. We currently don't support hidden communities on the client though
-  };
+  export type CommunityInfo = CommunityDetails &
+    WithPriority & {
+      pubkeyHex: string;
+    };
 
   export type LegacyGroupMemberInfo = {
     pubkeyHex: string;
     isAdmin: boolean;
   };
 
-  export type LegacyGroupInfo = {
+  type BaseUserGroup = WithPriority & {
+    joinedAtSeconds: number; // equivalent to the lastJoinedTimestamp in Session desktop but in seconds rather than MS
+    // NOTE Properties need to be optional going forward to backward compatibility
+    disappearingTimerSeconds?: number; // in seconds, 0 or undefined == disabled.
+  };
+
+  export type LegacyGroupInfo = BaseUserGroup & {
     pubkeyHex: string; // The legacy group "session id" (33 bytes).
     name: string; // human-readable; this should normally always be set, but in theory could be set to an empty string.
-    encPubkey: Uint8Array; // bytes (32 or empty)
+    encPubkey: Uint8ArrayLen32; // bytes (32 or empty)
     encSeckey: Uint8Array; // bytes (32 or empty)
-    priority: number; // -1 means hidden, 0 means normal, > 1 means pinned. We currently don't support hidden groups on the client though
+    // disappearingTimerSeconds: number; // in seconds, 0 == disabled.
```

Comment:
this might be an issue on my side

Reply:
actually no, I should just remove that line, as it is now part of `type BaseUserGroup`